darshan-util/pydarshan/tests/test_plot_exp_common.py

```diff
@@ -0,0 +1,172 @@
+            "POSIX",
+            plot_access_histogram,
+            [3, 17, 0, 0, 16, 0, 0, 0, 0, 0, 3, 4, 0, 0, 16, 0, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/ior_hdf5_example.darshan",
+            "MPI-IO",
+            plot_access_histogram,
+            [3, 17, 0, 0, 16, 0, 0, 0, 0, 0, 3, 4, 0, 0, 16, 0, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/sample-badost.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 131072, 0, 0, 0, 0],
+        ),
+        # subtle case where "POSIX" and "MPIIO"
+        # modules show different results
+        (
+            "tests/input/sample-dxt-simple.darshan",
+            "POSIX",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        ),
+        (
+            "tests/input/sample-dxt-simple.darshan",
+            "MPI-IO",
+            plot_access_histogram,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        ),
+        # more obvious case where "POSIX" and
+        # "MPIIO" modules show different results
+        (
+            None,
+            "POSIX",
+            plot_access_histogram,
+            [18, 2492, 14679, 0, 50486, 186, 0, 0, 0,
+            0, 43, 301, 2, 0, 50486, 0, 0, 0, 0, 0],
+        ),
+        (
+            None,
+            "MPI-IO",
+            plot_access_histogram,
+            [11, 2492, 2, 0, 0, 0, 0, 410, 86, 0, 2526,
+            303, 2, 0, 97812, 396, 0, 410, 86, 0],
+        ),
+        (
+            "examples/example-logs/dxt.darshan",
+            None,
+            plot_opcounts,
+            [6126, 1497, 264, 1379, 5597, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 39, 0, 1, 0, 0, 0, 0],
+        ),
+        (
+            "examples/example-logs/ior_hdf5_example.darshan",
+            None,
+            plot_opcounts,
+            [36, 23, 22, 4, 53, 0, 0, 36, 23, 1, 0, 0, 0,
+            0, 0, 0, 16, 0, 0, 0, 0, 0, 128, 1, 0, 0, 0, 9],
+        ),
+        (
+            "examples/example-logs/sample-badost.darshan",
+            None,
+            plot_opcounts,
+            [0, 131072, 2048, 2048, 131072, 0, 2048, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 34816, 97, 6144, 0, 0, 0, 2056],
+        ),
+        (
+            "examples/example-logs/sample.darshan",
+            None,
+            plot_opcounts,
+            [0, 16402, 2049, 0, 16404, 0, 0, 0, 18, 0, 0, 0, 0, 0,
+            0, 16384, 2048, 0, 0, 0, 0, 0, 74, 129, 0, 0, 0, 0],
+        ),
+    ],
+)
+@pytest.mark.skipif(not pytest.has_log_repo, # type: ignore
+                    reason="missing darshan_logs")
+@pytest.mark.parametrize("filename", ["imbalanced-io.darshan"])
+def test_bar_heights(log_path, mod, fig_func, expected_heights, select_log_repo_file):
+    # check bar graph heights
+
+    if log_path is None:
```

Comment:
This doesn't seem quite like the right design yet, since this test has a complex/stacked parametrization pattern.

- how do you add another case from the logs repo sustainably now that you've used `None` as a placeholder for "file in logs repo" in parameters above, without bolting on tons of extra logic in the test proper?

Perhaps something like this might be more sustainable (just add more logs repo filenames to params list when you want); I didn't add the extra test skipping logic needed from comment above

```diff
--- a/darshan-util/pydarshan/tests/test_plot_exp_common.py
+++ b/darshan-util/pydarshan/tests/test_plot_exp_common.py
@@ -59,7 +59,7 @@ def test_xticks_and_labels(log_path, func, expected_xticks, expected_xticklabels
 
 
 @pytest.mark.parametrize(
-    "log_path, mod, fig_func, expected_heights",
+    "filename, mod, fig_func, expected_heights",
     [
         (
             "examples/example-logs/dxt.darshan",
@@ -103,14 +103,14 @@ def test_xticks_and_labels(log_path, func, expected_xticks, expected_xticklabels
         # more obvious case where "POSIX" and
         # "MPIIO" modules show different results
         (
-            None,
+            "imbalanced-io.darshan",
             "POSIX",
             plot_access_histogram,
             [18, 2492, 14679, 0, 50486, 186, 0, 0, 0,
             0, 43, 301, 2, 0, 50486, 0, 0, 0, 0, 0],
         ),
         (
-            None,
+            "imbalanced-io.darshan",
             "MPI-IO",
             plot_access_histogram,
             [11, 2492, 2, 0, 0, 0, 0, 410, 86, 0, 2526,
@@ -148,13 +148,15 @@ def test_xticks_and_labels(log_path, func, expected_xticks, expected_xticklabels
 )
 @pytest.mark.skipif(not pytest.has_log_repo, # type: ignore
                     reason="missing darshan_logs")
-@pytest.mark.parametrize("filename", ["imbalanced-io.darshan"])
-def test_bar_heights(log_path, mod, fig_func, expected_heights, select_log_repo_file):
+def test_bar_heights(filename, mod, fig_func, expected_heights, select_log_repo_file):
     # check bar graph heights
 
-    if log_path is None:
+    logs_repo_path = select_log_repo_file
+    if logs_repo_path is not None:
         # for logs repo cases
         log_path = select_log_repo_file
+    else:
+        log_path = filename
 
     report = darshan.DarshanReport(log_path)
     fig, ax = plt.subplots()
```

Reply:
(note that `select_log_repo_file` will also fail catastrophically if the logs repo is absent, so careful with the skipping design on that point)

Reply:
As another hint, since this may be challenging to do elegantly, using `pytest.has_log_repo` in the test body proper, as part of conditional logic, may prove helpful for a sustainable design, possibly in addition to skips applied elsewhere, because the presence of the logs repo and working with a log repo file are two different things to check for.